boundary_alinged_remesh.py

```diff
@@ -15,173 +15,189 @@
             if edge.calc_length() < lower_length and not edge.is_boundary:
                 verts = set(edge.verts)
                 if verts & lock_verts:
                     continue
                 collapse.append(edge)
                 lock_verts |= verts
-        
+
         bmesh.ops.collapse(self.bm, edges=collapse)
         bmesh.ops.beautify_fill(self.bm, faces=self.bm.faces, method="ANGLE")
-    
+
     def align_verts(self, rule=(-1, -2, -3, -4)):
         # Align verts to the nearest boundary by averaging neigbor vert locations selected
         # by a specific rule,
-        
+
         # Rules work by sorting edges by angle relative to the boundary.
         # Eg1. (0, 1) stands for averagiing the biggest angle and the 2nd biggest angle edges.
         # Eg2. (-1, -2, -3, -4), averages the four smallest angle edges
         for vert in self.bm.verts:
             if not vert.is_boundary:
                 vec = self.nearest_boundary_vector(vert.co)
                 neighbor_locations = [edge.other_vert(vert).co for edge in vert.link_edges]
-                best_locations = sorted(neighbor_locations, 
-                                        key = lambda n_loc: abs((n_loc - vert.co).normalized().dot(vec)))
+                best_locations = sorted(neighbor_locations,
+                                        key=lambda n_loc: abs((n_loc - vert.co).normalized().dot(vec)))
                 co = vert.co.copy()
                 le = len(vert.link_edges)
-                for i in   rule:
+                for i in rule:
                     co += best_locations[i % le]
                 co /= len(rule) + 1
                 co -= vert.co
                 co -= co.dot(vert.normal) * vert.normal
                 vert.co += co
-    
+
     def reproject(self):
         """ Recovers original shape """
         for vert in self.bm.verts:
             location, normal, index, dist = self.bvh.find_nearest(vert.co)
             if location:
                 vert.co = location
-    
-    def remesh(self,edge_length=0.05, iterations=30, quads=True):
+
+    def remesh(self, edge_length=1.0, iterations=30, quads=True):
         """ Coordenates remeshing """
         if quads:
-            rule = (-1,-2, 0, 1)
+            rule = (-1, -2, 0, 1)
         else:
             rule = (0, 1, 2, 3)
-        
+
         for _ in range(iterations):
             self.enforce_edge_length(edge_length=edge_length)
             self.align_verts(rule=rule)
             self.reproject()
-        
+
         if quads:
             bmesh.ops.join_triangles(self.bm, faces=self.bm.faces,
                                      angle_face_threshold=3.14,
                                      angle_shape_threshold=3.14)
         return self.bm
 
+
 class Remesher(bpy.types.Operator):
     bl_idname = "remesh.boundary_aligned_remesh"
     bl_label = "Boundary Aligned Remesh"
     bl_options = {"REGISTER", "UNDO"}
-    
+
+    def __init__(self):

```

Comment:
Nice but wouldn't be better to make it a `invoke`  method instead of `__init__` and add a `self.execute(context)` call at the end?

Reply:
As I know invoke is for modal operators, and I use init for the properties default value. maybe I'm wrong

Reply:
in the api it says:

> `Operator.invoke` is used to initialize the operator from the context at the moment the operator is called. invoke() is typically used to assign properties which are then used by execute(). Some operators don’t have an execute() function, removing the ability to be repeated from a script or macro.
https://docs.blender.org/api/blender2.8/bpy.types.Operator.html

Looks like an `__init__` is fine but to be honest, I never saw an operator using it, so I am being kinda skeptical.

Reply:
For sure it could be invoke.